develop/tutorials/articles/03-developing-a-web-application/04-writing-web-application-liferay/01-writing-a-web-application.markdown

```diff
@@ -0,0 +1,635 @@
+
+![Figure x: This is the Guestbook application's form for adding entries.](../../images/first-guestbook-portlet.png)
+
+Implementing the portlet action (what happens when the user clicks *Save*) is
+your next task.
+
+### Implementing Portlet Actions
+
+When users submit the form, your application needs to store the form data so it
+can be displayed in the guestbook. To keep this first application simple, you'll
+implement this using a part of the Portlet API called Portlet Preferences.
+Normally, of course, you'd use a database, and Liferay makes it very easy to
+work with databases using its Service Builder framework. For now, however, you
+can create the first iteration of your guestbook application using portlet
+preferences.
+
+To make your portlet do anything other than re-render itself, you must
+implement portlet actions. An action defines some processing, usually based on
+user input, that the portlet must perform before it renders itself. In the case
+of the guestbook portlet, the action you need to implement saves a guestbook
+entry that a user typed into the form. Saved guestbook entries can be retrieved
+and displayed later.
+
+Since you're using Liferay's MVC Portlet framework, you have an easy way to
+implement actions. Portlet actions can be implemented in the portlet class,
+which acts as the controller. In the form you just created, you made an action
+URL, and you called it `addEntry`. To create a portlet action, you create a
+method in the portlet class with the same name. `MVCPortlet` calls that method
+when a user triggers its matching URL.
+
+1. Open the `GuestbookPortlet.java`. Blade Tools generated this class when you
+   created the portlet project.
+
+2. Create a method with the following signature:
+
+        public void addEntry(ActionRequest request, ActionResponse response) {
+
+        }
+
+3. Add the required `javax.portlet.ActionRequest` and
+   `javax.portlet.ActionResponse` imports.
+
+You've now created a portlet action. It doesn't do anything, but at least you
+won't get an error now if you submit your form. Next, you should make the
+action save the form data.
+
+Because of the limitations of the portlet preferences API, you need to store
+each guestbook entry as a `String` in a string array. Since you have two fields
+in your form, you need to have a delimiter so you can determine where the user
+name ends and the guestbook entry begins. The caret symbol (`^`) makes a good
+delimiter because users are highly unlikely to use that symbol in a guestbook
+entry.
+
++$$$
+
+**Note:** Clearly, using portlet preferences and string delimiters is not the
+best way to implement a mechanism for saving guestbook entries. This approach
+is easy to implement but it's really just a shortcut that should only be
+considered when developing a prototype application. To learn how to implement a
+robust mechanism for saving application data, follow the *Writing a Data-Driven
+Application* learning path.
+
+$$$
+
```

Comment:
I thought we were going to use the Configuration API instead? 

https://dev.liferay.com/develop/tutorials/-/knowledge_base/7-0/transitioning-from-portlet-preferences-to-the-configuration-api